syntax = "proto3";

package synapse;

enum SpikeDetectMode {
  kThreshold = 0;
  kTemplate = 1;
  kWavelet = 2;
}

message SpikeDetectOptions {
  repeated SpikeDetectMode mode = 1;
}

message SpikeDetectConfig {
  SpikeDetectMode mode = 1;
  uint32 threshold_uV = 2;
  repeated uint32 template_uV = 3;
  bool sort = 4;
}
